test/pkg/event/event.go

```diff
@@ -1,12 +1,305 @@
 package event
 
 import (
+	"context"
+	"fmt"
 	"os"
 	"strconv"
+	"strings"
+	"time"
+
+	"github.com/openshift/ptp-operator/test/pkg/client"
+	"github.com/openshift/ptp-operator/test/pkg/namespaces"
+	"github.com/openshift/ptp-operator/test/pkg/pods"
+	"github.com/openshift/ptp-operator/test/pkg/testconfig"
+	lib "github.com/redhat-cne/ptp-listener-lib"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+)
+
+const (
+	ConsumerSidecarTestNamespace  = "ptp-event-framework-test"
+	ProviderSidecarTestNamespace  = "openshift-ptp"
+	TestSidecarName               = "cloud-proxy-sidecar"
+	TestSidecarContainerName      = "sidecar"
+	TestSidecarSaName             = "sidecar-sa"
+	TestServiceName               = "consumer-events-subscription-service"
+	TestSidecarSuccessLogString   = "rest service returned healthy status"
+	CustomerCloudEventProxyPort   = 8089
+	ProviderCloudEventProxyPort   = 9085
+	sidecarNamespaceDeleteTimeout = time.Minute * 2
 )
 
-// Enabled event if ptp event is required
+// enables event if ptp event is required
 func Enable() bool {
 	eventMode, _ := strconv.ParseBool(os.Getenv("ENABLE_PTP_EVENT"))
 	return eventMode
 }
+
+// create ptp-events sidecar
+func CreateEventProxySidecar(nodeName string) error {
+	rootUser := pointer.Int64(0)
+	aPod := corev1.Pod{
+		TypeMeta: v1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Name:      "cloud-proxy-sidecar",
+			Namespace: ConsumerSidecarTestNamespace,
+			Labels: map[string]string{
+				"app.kubernetes.io/name": "proxy",
+			},
+		},
+		Spec: corev1.PodSpec{
+			Volumes: []corev1.Volume{
+				{
+					Name: "pubsubstore",
+					VolumeSource: corev1.VolumeSource{
+						EmptyDir: &corev1.EmptyDirVolumeSource{},
+					},
+				},
+			},
+			ServiceAccountName: TestSidecarSaName,
+			Containers: []corev1.Container{
+				{Name: TestSidecarContainerName,
+					Image: "quay.io/redhat-cne/cloud-event-proxy",
+					SecurityContext: &corev1.SecurityContext{
+						Privileged: pointer.Bool(true),
+						RunAsUser:  rootUser,
+					},
+
+					Args: []string{"--metrics-addr=127.0.0.1:9091",
+						"--store-path=/store",
+						"--transport-host=consumer-events-subscription-service." + ConsumerSidecarTestNamespace + ".svc.cluster.local:9043",
+						"--http-event-publishers=ptp-event-publisher-service-" + nodeName + "." + ProviderSidecarTestNamespace + ".svc.cluster.local:9043",
```

Comment:
is this full node name or just the first part?

Reply:
Fixed. Thanks, good catch!